pkg/api/persistentvolume/util.go

```diff
@@ -33,12 +33,5 @@ func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.Persist
 }
 
 func hasExpansionSecrets(oldPVSpec *api.PersistentVolumeSpec) bool {
-	if oldPVSpec == nil || oldPVSpec.CSI == nil {
-		return false
-	}
-
-	if oldPVSpec.CSI.ControllerExpandSecretRef != nil {
```

Comment:
Do these `if` statements represent different `concepts`? If yes, I think I'm in favor of keeping them separate. However, I think we can get rid of the `return false` in favor of `return oldPVSpec...`

I think my opinions here apply to much of the `if` statements in this pr. In general, I support separate if statements for separate concepts, and avoiding `return (true|false)` when its possible to return the results of an expression instead.

Reply:
When I take a looked that code as an outsider, I assume the purpose that if statements to make nil controls. It tries to check the most inner variable, if one of the variable in that path is nil, it returns false. So in my opinion these if statements is not represent different concepts.  To return true there is one condition and I guess it makes if statements to serve one concept.

Also I agree with you in `support separate if statements for separate concepts`.

If I'm wrong, or missing some point, I want to hear opinions.